contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -848,11 +848,23 @@ pltsql_GetNewTempObjectId()
 			 * GetNewObjectId will return the right value on the next call.  
 			 */
 			if (tempOidStart < FirstNormalObjectId)
+			{
+				/* 
+				* This situation should not be reached in an ideal state since oid < FirstNormalObjectId is during bootstrapping
+				* However, if it does we simply start from FirstNormalObjectId 
+				*/
+				elog(NOTICE, "TransamVariables->nextOid is below FirstNormalObjectId");
 				tempOidStart = FirstNormalObjectId;
+			}
 
 			/* If the OID range would wraparound, start from beginning instead. */
 			if (tempOidStart + temp_oid_buffer_size < tempOidStart)
 			{
+				/* Raise a notice */
+				elog(NOTICE, "OID range wraparound reached while trying to allocate OIDs for tsql temp objects");
```

Comment:
This is fine to log, but let's have a clearer message here. If the next OID is within `temp_oid_buffer_size` of the end of the buffer, the end of the buffer would be in wraparound territory (ie < FirstNormalObjectId). So we must move the buffer range to the beginning of the buffer instead. Let's shorten that explanation and provide that as the log.